packages/neon_framework/lib/src/blocs/user_status.dart

```diff
@@ -89,18 +142,103 @@ class _UserStatusBloc extends InteractiveBloc implements UserStatusBloc {
         data = response.body.ocs.data;
       }
 
-      updateStatus(username, Result.success(data));
-    } catch (e, s) {
+      updateStatus(username, Result.success(correctStatus(data)));
+    } on DynamiteStatusCodeException catch (e, s) {
       debugPrint(e.toString());
       debugPrint(s.toString());
       updateStatus(username, Result.error(e));
     }
   }
 
-  void updateStatus(String username, Result<user_status.$PublicInterface> result) {
+  /// Reconstructs the message and icon for a private [status] using the [predefinedStatuses] in case it is a predefined status.
+  user_status.$PublicInterface correctStatus(user_status.$PublicInterface status) {
+    var correctedResult = status;
+    if (correctedResult is user_status.Private && correctedResult.messageIsPredefined) {
+      final id = correctedResult.messageId;
+      final predefinedStatus =
+          predefinedStatuses.valueOrNull?.data?.firstWhereOrNull((predefined) => predefined.id == id);
+
+      if (predefinedStatus != null) {
+        correctedResult = correctedResult.rebuild(
+          (b) => b
+            ..message = predefinedStatus.message
+            ..icon = predefinedStatus.icon,
+        );
+      }
+    }
+
+    return correctedResult;
```

Comment:
```suggestion
    if (status is user_status.Private && status.messageIsPredefined) {
      final id = status.messageId;
      final predefinedStatus =
          predefinedStatuses.valueOrNull?.data?.firstWhereOrNull((predefined) => predefined.id == id);

      if (predefinedStatus != null) {
        return status.rebuild(
          (b) => b
            ..message = predefinedStatus.message
            ..icon = predefinedStatus.icon,
        );
      }
    }

    return status;
```

Sorry for the back and forth